packages/@aws-cdk/aws-chatbot/test/slack-channel-configuration.test.ts

```diff
@@ -138,6 +140,47 @@ describe('SlackChannelConfiguration', () => {
     });
   });
 
+  test('specifying log retention', () => {
+    new chatbot.SlackChannelConfiguration(stack, 'MySlackChannel', {
+      slackWorkspaceId: 'ABC123',
+      slackChannelId: 'DEF456',
+      slackChannelConfigurationName: 'ConfigurationName',
+      logRetention: logs.RetentionDays.ONE_MONTH,
+    });
+
+    expect(stack).toHaveResourceLike('Custom::LogRetention', {
+      LogGroupName: '/aws/chatbot/ConfigurationName',
+      RetentionInDays: 30,
+      LogGroupRegion: 'us-east-1',
+    });
+  });
+
+  test('getting configuration metric', () => {
+    const slackChannel = new chatbot.SlackChannelConfiguration(stack, 'MySlackChannel', {
+      slackWorkspaceId: 'ABC123',
+      slackChannelId: 'DEF456',
+      slackChannelConfigurationName: 'ConfigurationName',
+      logRetention: logs.RetentionDays.ONE_MONTH,
+    });
+
+    expect(slackChannel.metric('MetricName')).toEqual(new cloudwatch.Metric({
```

Comment:
Would be better assertions if you could create a CF resource from this metric and assert these values using stack inspection APIs, like `toHaveResourceLike()`.

How about creating an alarm from this metric and asserting these very values but on the `AWS::CloudWatch::Alarm` resource?

Reply:
@nija-at Actually, similar to my **Other** question on https://github.com/aws/aws-cdk/issues/10135 about the inability to expose a `LogGroup`, I don't think this will work with [`AWS::CloudWatch::Alarm`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cw-alarm.html) as it doesn't have the concept of a region.

I think there's a general problem here when it comes to global AWS services like Chatbot that write CloudWatch resources to `us-east-1` and the lack of classes/constructs that represent those cloudwatch resources in CDK.

Reply:
I refrained from exposing a `logGroup()` method or even a `LogGroupArn` because of that.
I think the metrics are potentially usable in [dashboards](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/Cross-Account-Cross-Region.html#enable-cross-account-cross-Region) but am not sure if we'd want to test using that instead.

Reply:
What do you think?

Reply:
I tried with the dashboard and the test would be something like

```
  test('getting configuration metric', () => {
    const slackChannel = new chatbot.SlackChannelConfiguration(stack, 'MySlackChannel', {
      slackWorkspaceId: 'ABC123',
      slackChannelId: 'DEF456',
      slackChannelConfigurationName: 'ConfigurationName',
      logRetention: logs.RetentionDays.ONE_MONTH,
    });
    const metric = slackChannel.metric('MetricName');
    new cloudwatch.Dashboard(stack, 'Dashboard', { dashboardName: 'DashboardName' })
      .addWidgets(new cloudwatch.GraphWidget({ left: [metric] }));

    expect(metric).toEqual(new cloudwatch.Metric({
      namespace: 'AWS/Chatbot',
      region: 'us-east-1',
      dimensions: {
        ConfigurationName: 'ConfigurationName',
      },
      metricName: 'MetricName',
    }));
    expect(stack).toHaveResourceLike('AWS::CloudWatch::Dashboard', {
      DashboardBody: {
        'Fn::Join': [
          '',
          [
            '{"widgets":[{"type":"metric","width":6,"height":6,"x":0,"y":0,"properties":{"view":"timeSeries","region":"',
            {
              Ref: 'AWS::Region',
            },
            '","metrics":[["AWS/Chatbot","MetricName","ConfigurationName","ConfigurationName",{"region":"us-east-1"}]],"yAxis":{}}}]}',
          ],
        ],
      },
      DashboardName: 'DashboardName',
    });
  });
```

But my feeling is that this will be a fragile test heavily dependent on how `GraphWidget` is rendered... if any property changes position, the test will likely break.

Reply:
I wanted to go back to your first response.

> @nija-at Actually, similar to my Other question on #10135 about the inability to expose a LogGroup, I don't think this will work with `AWS::CloudWatch::Alarm` as it doesn't have the concept of a region.

While there might be a problem when deployed, it should still be good enough for a unit test, no? We're only looking to synthesize and assert.
Is there any reason why CDK will reject when `new cloudwatch.Alarm()` is used?

Reply:
It would definitely work as a unit test for assertion purposes.
My only qualm with it is that unit tests can in a way show how code should be used and we know that this not how this method should be used.

If you think it's fine and that it won't be read for more than assertion purposes, I am happy to update the PR

Reply:
I think this is fine. We'll need to look into the issue of deployability you've raised separately.

Reply:
Cool... I'll have an updated PR soon

Reply:
Updated